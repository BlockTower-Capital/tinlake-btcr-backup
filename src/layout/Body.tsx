import React, { FunctionComponent } from 'react';
import { Box } from 'grommet';
import { Route, Switch } from 'react-router';
import invoiceRoutes from '../invoices/routes';
import orderRoutes from '../orders/routes';
import contactsRoutes from '../contacts/routes';
import Invoices from '../invoices/Invoices';
import CreateInvoice from '../invoices/CreateInvoice';
import Orders from '../orders/Orders';
import Contacts from '../contacts/Contacts';

const Body: FunctionComponent = () => (
  <Box justify="center" direction="row" fill="true" background="#f9f9fa">
    <Box width="xlarge">
      <Switch>
        <Route path={invoiceRoutes.new} component={CreateInvoice} />
        <Route path={invoiceRoutes.base} component={Invoices} />
        <Route path={orderRoutes.base} component={Orders} />
        <Route path={contactsRoutes.base} component={Contacts} />
      </Switch>
    </Box>
  </Box>
);

Body.displayName = 'Body';

export default Body;
