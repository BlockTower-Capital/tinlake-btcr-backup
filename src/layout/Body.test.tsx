import React from 'react';
import { shallow } from 'enzyme';

import Body from './Body';

describe('Body', () => {
  it('matches snapshot', () => {
    const bodyShallow = shallow(<Body />);
    expect(bodyShallow).toMatchSnapshot();
  });
});
