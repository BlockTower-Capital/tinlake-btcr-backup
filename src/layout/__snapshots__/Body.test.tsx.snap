// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body matches snapshot 1`] = `
<Box
  align="center"
  direction="row"
  gridArea="main"
  justify="center"
>
  <Switch>
    <Route
      component={[Function]}
      path="/invoices"
    />
    <Route
      component={[Function]}
      path="/orders"
    />
    <Route
      component={[Function]}
      path="/contacts"
    />
  </Switch>
</Box>
`;
